src/test/java/com/it4045/pokemonviewer/enterprise/EnterpriseApplicationTests.java

```diff
@@ -42,6 +42,20 @@ void fetchPokemonByName_returnsCharizardId() throws IOException {
         thenReturnOnePokemonCharziardWithIdOne();
     }
 
+    @Test
+    void fetchPokemonByNumber_returns() throws IOException {
+        givenPokemonDataIsReadilyAvailable();
+        whenSearchPokemonWithNumberThree();
+        thenReturnOnePokemonWithNumberThree();
+    }
+
+    private void whenSearchPokemonWithNumberThree() { pokemon = pokemonService.fetchByNumber(3); }
+
+    private void thenReturnOnePokemonWithNumberThree() {
+        String pokemonNumber = pokemon.getPokemonNumber();
+        assertEquals("3", pokemonNumber);
+    }
+
```

Comment:
Third technical suggestion: Added and implemented a test that makes use of this new functionality.

I created a test that will determine if Pokemon can be retrieved by their number, along with the functions required to implement this test, which interact with the earlier changes that I made in the Pokemon service class. This test, along with all of the others that have been created so far, was successful when I ran it.